In [2]:
import pandas as pd
import numpy as np
fulldata = pd.read_csv("/Users/nicolasst-germain/Documents/GitHub/audit_google_news/all_data_saisie/donneesBrutesNStG.csv")

/Users/nicolasst-germain/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pandas/compat/__init__.py:124: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


In [3]:
fulldata

Villes Mots-clés  position du média  \
0                Longueuil  accident                  1   
1            Rouyn-Noranda  accident                  1   
2                Longueuil  accident                  2   
3            Rouyn-Noranda  accident                  2   
4            Rouyn-Noranda  accident                  3   
...                    ...       ...                ...   
488021  Dolbeau-Mistassini   déficit                  4   
488022  Dolbeau-Mistassini   déficit                  5   
488023  Dolbeau-Mistassini   déficit                  6   
488024  Dolbeau-Mistassini   déficit                  7   
488025  Dolbeau-Mistassini   déficit                  8   

                   Nom du média  \
0        Le Journal de Montréal   
1                  Radio-Canada   
2                   iHeartRadio   
3                     La Presse   
4                   iHeartRadio   
...                         ...   
488021  Le Quotidien (Saguenay)   
488022  Le Quotidien (Saguenay)   
488023  Le Quotidien (Saguenay)   
488024  Le Quotidien (Saguenay)   
488025  Le Quotidien (Saguenay)   

                                       Titre de l'article Date de publication  \
0       Longueuil: un accident de voiture force la fer...          2021-07-27   
1       Une nouvelle voie de dépassement sur la route 117          2021-07-07   
2       Accident à Longueuil: un automobiliste de 22 a...          2021-07-17   
3       Un mort et 2 blessés dans deux accidents à Rou...          2020-12-09   
4            Un motocycliste perd la vie sur la route 117          2021-07-12   
...                                                   ...                 ...   
488021  Un parc multisport de 3,3 M$ sur l'ancien terr...          2021-09-16   
488022  Dolbeau-Mistassini inaugure son café libre-ser...          2021-09-24   
488023  Dolbeau-Mistassini bonifie ses programmes d'ai...          2021-07-13   
488024  Dolbeau-Mistassini veut développer son autonom...          2021-02-15   
488025  La bureaucratie empêche une famille belge de v...          2021-10-03   

        Journée de différence  \
0                         7.0   
1                        27.0   
2                        17.0   
3                       237.0   
4                        22.0   
...                       ...   
488021                   93.0   
488022                   85.0   
488023                  158.0   
488024                  306.0   
488025                   76.0   

                                                      URL  \
0       https://www.journaldemontreal.com/2021/07/26/l...   
1       https://ici.radio-canada.ca/nouvelle/1807406/n...   
2       https://www.iheartradio.ca/energie/energie-mon...   
3       https://www.lapresse.ca/actualites/justice-et-...   
4       https://www.iheartradio.ca/energie/energie-val...   
...                                                   ...   
488021  https://www.lequotidien.com/2021/09/16/un-parc...   
488022  https://www.lequotidien.com/2021/09/24/dolbeau...   
488023  https://www.lequotidien.com/2021/07/13/dolbeau...   
488024  https://www.lequotidien.com/2021/02/16/dolbeau...   
488025  https://www.lequotidien.com/2021/10/03/la-bure...   

       date et heure saisie type mot-clés                  media_url  \
0       2021-08-03 08:00:09         local  www.journaldemontreal.com   
1       2021-08-03 08:00:09         local        ici.radio-canada.ca   
2       2021-08-03 08:00:09         local         www.iheartradio.ca   
3       2021-08-03 08:00:09         local            www.lapresse.ca   
4       2021-08-03 08:00:09         local         www.iheartradio.ca   
...                     ...           ...                        ...   
488021  2021-12-18 17:53:23      national        www.lequotidien.com   
488022  2021-12-18 17:53:23      national        www.lequotidien.com   
488023  2021-12-18 17:53:23      national        www.lequotidien.com   
488024  2021-12-18 17:53:23      national        www.lequotidien.co

In [5]:
fulldata.columns = ['Villes', 'MotsClés', 'PositionMédia', 'NomMédia',
       'TitreArticle', 'DaterPublication', 'JournéeDifférence',
       'URL', 'DateHeureSaisie', 'TypeMotClés', 'MediaUrl',
       'TypeMedia', 'VilleSiègeSocial', 'domaine']

In [81]:
fulldata.dtypes

Villes                object
MotsClés              object
PositionMédia          int64
NomMédia              object
TitreArticle          object
DaterPublication      object
JournéeDifférence    float64
URL                   object
DateHeureSaisie       object
TypeMotClés           object
MediaUrl              object
TypeMedia             object
VilleSiègeSocial      object
domaine               object
dtype: object

In [53]:
fulldata

Villes  MotsClés  PositionMédia                 NomMédia  \
0                Longueuil  accident              1   Le Journal de Montréal   
1            Rouyn-Noranda  accident              1             Radio-Canada   
2                Longueuil  accident              2              iHeartRadio   
3            Rouyn-Noranda  accident              2                La Presse   
4            Rouyn-Noranda  accident              3              iHeartRadio   
...                    ...       ...            ...                      ...   
488021  Dolbeau-Mistassini   déficit              4  Le Quotidien (Saguenay)   
488022  Dolbeau-Mistassini   déficit              5  Le Quotidien (Saguenay)   
488023  Dolbeau-Mistassini   déficit              6  Le Quotidien (Saguenay)   
488024  Dolbeau-Mistassini   déficit              7  Le Quotidien (Saguenay)   
488025  Dolbeau-Mistassini   déficit              8  Le Quotidien (Saguenay)   

                                             TitreArticle DaterPublication  \
0       Longueuil: un accident de voiture force la fer...       2021-07-27   
1       Une nouvelle voie de dépassement sur la route 117       2021-07-07   
2       Accident à Longueuil: un automobiliste de 22 a...       2021-07-17   
3       Un mort et 2 blessés dans deux accidents à Rou...       2020-12-09   
4            Un motocycliste perd la vie sur la route 117       2021-07-12   
...                                                   ...              ...   
488021  Un parc multisport de 3,3 M$ sur l'ancien terr...       2021-09-16   
488022  Dolbeau-Mistassini inaugure son café libre-ser...       2021-09-24   
488023  Dolbeau-Mistassini bonifie ses programmes d'ai...       2021-07-13   
488024  Dolbeau-Mistassini veut développer son autonom...       2021-02-15   
488025  La bureaucratie empêche une famille belge de v...       2021-10-03   

        JournéeDifférence                                                URL  \
0                     7.0  https://www.journaldemontreal.com/2021/07/26/l...   
1                    27.0  https://ici.radio-canada.ca/nouvelle/1807406/n...   
2                    17.0  https://www.iheartradio.ca/energie/energie-mon...   
3                   237.0  https://www.lapresse.ca/actualites/justice-et-...   
4                    22.0  https://www.iheartradio.ca/energie/energie-val...   
...                   ...                                                ...   
488021               93.0  https://www.lequotidien.com/2021/09/16/un-parc...   
488022               85.0  https://www.lequotidien.com/2021/09/24/dolbeau...   
488023              158.0  https://www.lequotidien.com/2021/07/13/dolbeau...   
488024              306.0  https://www.lequotidien.com/2021/02/16/dolbeau...   
488025               76.0  https://www.lequotidien.com/2021/10/03/la-bure...   

            DateHeureSaisie TypeMotClés                   MediaUrl  \
0       2021-08-03 08:00:09       local  www.journaldemontreal.com   
1       2021-08-03 08:00:09       local        ici.radio-canada.ca   
2       2021-08-03 08:00:09       local         www.iheartradio.ca   
3       2021-08-03 08:00:09       local            www.lapresse.ca   
4       2021-08-03 08:00:09       local         www.iheartradio.ca   
...                     ...         ...                        ...   
488021  2021-12-18 17:53:23    national        www.lequotidien.com   
488022  2021-12-18 17:53:23    national        www.lequotidien.com   
488023  2021-12-18 17:53:23    national        www.lequotidien.com   
488024  2021-12-18 17:53:23    national        www.lequotidien.com   
488025  2021-12-18 17:53:23    national        www.lequotidien.com   

             TypeMedia VilleSiègeSocial                domaine  
0             national         Montréal  journaldemontreal.com  
1       local/regional         Montréal    ici.radio-canada.ca  
2            non media         Montréal     www.iheartradio.ca  
3             national         Montréal            lapresse.

In [94]:
fulldata.describe()

PositionMédia  JournéeDifférence
count  488026.000000      448534.000000
mean       12.678650         128.022143
std         7.226784         190.986844
min         1.000000           0.000000
25%         6.000000          25.000000
50%        13.000000          84.000000
75%        19.000000         171.000000
max        25.000000        5715.000000

Pourrait être intéressant pour savoir la fréquence de mise à jour des articles mis en ligne sur Google News. On remarque que la fréquence est variable. 

# EXPLORATION DES DONNÉES GÉNÉRALES

In [55]:
fulldata["TypeMedia"].value_counts()

local/regional    297109
national          140304
non media          25828
hors Québec        24785
Name: TypeMedia, dtype: int64

Beaucoup plus de médias locaux.

In [56]:
fulldata["TypeMedia"].count()

488026

In [57]:
fulldata["VilleSiègeSocial"].value_counts()

Montréal                      187573
Québec                         47695
hors Québec                    22645
Trois-Rivières                 21500
Sherbrooke                     16622
                               ...  
Kirkland                          33
Saint-Bruno-de-Montarville        20
Coteau-du-Lac                     11
Chandler                           4
Chénéville                         2
Name: VilleSiègeSocial, Length: 75, dtype: int64

Montréal est sur-représenté dans les résultats. MAIS la-dedans tous les radio-canada sont considérés comme Montréal dans le siège social. Alors qu'il y a certainement des médias locaux là-dedans ou du moins des contenus produit localement. Certes, leurs terre d'attache est Montréal. 

In [58]:
fulldata["domaine"].value_counts()

ici.radio-canada.ca                 82580
lapresse.ca                         30034
journaldemontreal.com               29077
journaldequebec.com                 25843
lesoleil.com                        20709
                                    ...  
www.nouvellespublications.com           1
fr.blastingnews.com                     1
www.meyomessalainternational.com        1
www.cyclismactu.net                     1
www.actugaming.net                      1
Name: domaine, Length: 1444, dtype: int64

In [59]:
top20 = fulldata["domaine"].value_counts().head(20)
top20

ici.radio-canada.ca      82580
lapresse.ca              30034
journaldemontreal.com    29077
journaldequebec.com      25843
lesoleil.com             20709
lenouvelliste.ca         17967
tvanouvelles.ca          16751
latribune.ca             15809
necrocanada.com          12001
ledroit.com              11610
lequotidien.com          10018
ledevoir.com              9809
lavoixdelest.ca           9150
lanouvelle.net            8477
cimtchau.ca               7762
www.iheartradio.ca        6878
infodimanche.com          6628
meteomedia.com            4628
lemanic.ca                3724
journalacces.ca           3679
Name: domaine, dtype: int64

In [60]:
fulldata.count()

Villes               488026
MotsClés             488026
PositionMédia        488026
NomMédia             488026
TitreArticle         488026
DaterPublication     448534
JournéeDifférence    448534
URL                  488026
DateHeureSaisie      488026
TypeMotClés          488026
MediaUrl             488026
TypeMedia            488026
VilleSiègeSocial     479654
domaine              488026
dtype: int64

In [61]:
fulldata["Villes"].value_counts()

Montréal_V2                        5200
Québec                             5189
Trois-Rivières                     5175
Montréal                           5172
Rimouski                           5148
                                   ... 
Ormstown                           3342
Boischatel                         3327
Mansfield                          3217
Chisasibi                          3134
Saint-Pierre-de-l'Île-d'Orléans    2172
Name: Villes, Length: 105, dtype: int64

^^ On voit dans ce tableau que seul Montréal_V2 compte 100% des résultats. Le plus bas est Saint-Pierre-de-l'Île-d'Orléans. Montréal_V2 est aussi le seul dont la recherche ne contenait pas le "près de ___"

# Détailler Radio-Canada et TVA

In [62]:
fulldata["domaine"].str.contains("ici.radio-canada.ca").value_counts()

False    405446
True      82580
Name: domaine, dtype: int64

In [63]:
radcan = fulldata.domaine.str.contains("ici.radio-canada.ca", na = False)
fulldata[radcan]

Villes            MotsClés  PositionMédia      NomMédia  \
1            Rouyn-Noranda            accident              1  Radio-Canada   
9            Rouyn-Noranda            accident              5  Radio-Canada   
11           Rouyn-Noranda            accident              6  Radio-Canada   
15           Rouyn-Noranda            accident              8  Radio-Canada   
17           Rouyn-Noranda            accident              9  Radio-Canada   
...                    ...                 ...            ...           ...   
487924           Coaticook      Biden OR Trump              1  Radio-Canada   
487932           Coaticook             déficit              2  Radio-Canada   
487941     Cookshire-Eaton  "premier ministre"              2  Radio-Canada   
487998  Dolbeau-Mistassini  "premier ministre"             17  Radio-Canada   
488010  Dolbeau-Mistassini         immigration              7  Radio-Canada   

                                             TitreArticle DaterPublication  \
1       Une nouvelle voie de dépassement sur la route 117       2021-07-07   
9       Accident à Granada : les proches des victimes ...       2021-04-30   
11            Une collision sur la route 101 fait un mort       2020-11-02   
15      La CSN réclame une enquête sur un accident mor...       2021-06-03   
17      Accident mortel à Granada : Mathieu Desgagné e...       2021-04-23   
...                                                   ...              ...   
487924  Pourquoi couvrir les 100 premiers jours d'un p...       2021-04-26   
487932  Le CISSS des Îles anticipe un déficit de 4,8 M...       2021-05-27   
487941  Fraude de 2,2 M$ : 6 ans de prison pour une an...       2021-09-24   
487998  Deux cas de COVID-19 répertoriés dans les écol...       2021-09-13   
488010  Le Lac-Saint-Jean, ses enjeux oubliés pendant ...       2021-09-19   

        JournéeDifférence                                                URL  \
1                    27.0  https://ici.radio-canada.ca/nouvelle/1807406/n...   
9                    95.0  https://ici.radio-canada.ca/nouvelle/1789475/m...   
11                  274.0  https://ici.radio-canada.ca/nouvelle/1746233/c...   
15                   61.0  https://ici.radio-canada.ca/nouvelle/1798647/a...   
17                  102.0  https://ici.radio-canada.ca/nouvelle/1787303/a...   
...                   ...                                                ...   
487924              236.0  https://ici.radio-canada.ca/nouvelle/1787360/c...   
487932              205.0  https://ici.radio-canada.ca/nouvelle/1796589/d...   
487941               85.0  https://ici.radio-canada.ca/nouvelle/1826901/f...   
487998               96.0  https://ici.radio-canada.ca/nouvelle/1823852/p...   
488010               90.0  https://ici.radio-canada.ca/nouvelle/1825456/l...   

            DateHeureSaisie TypeMotClés             MediaUrl       TypeMedia  \
1       2021-08-03 08:00:09       local  ici.radio-canada.ca  local/regional   
9       2021-08-03 08:00:10       local  ici.radio-canada.ca  local/regional   
11      2021-08-03 08:00:10       local  ici.radio-canada.ca  local/regional   
15      2021-08-03 08:00:10       local  ici.radio-canada.ca  local/regional   
17      2021-08-03 08:00:10       local  ici.radio-canada.ca  local/regional   
...                     ...         ...                  ...             ...   
487924  2021-12-18 17:49:37    national  ici.radio-canada.ca  local/regional   
487932  2021-12-18 17:49:44    national  ici.radio-canada.ca  local/regional   
487941  2021-12-18 17:49:51    national  ici.radio-canada.ca  local/regional   
487998  2021-12-18 17:52:17    national  ici.radio-canada.ca  local/regional   
488010  2021-12-18 17:52:58    national  ici.radio-canada.ca  local/regional   

       VilleSiègeSocial              domaine  
1              Montréal  ici.radio-canada.ca  
9              Montréal  ici.radio-canada.ca  
11             Montréal  ici.radio-canada.ca  
15             Montréal 

In [65]:
fulldata[radcan].TypeMedia.value_counts()

local/regional    65283
national          14744
hors Québec        2553
Name: TypeMedia, dtype: int64

Pour Rad-Can y'a bcp de contenu produit par local/régional et peu par National. Cela a été déterminé par un script produit par JHR. 

In [67]:
tva = fulldata.domaine.str.contains("tvanouvelles.ca", na = False)
fulldata[tva].TypeMedia.value_counts()


national          12701
local/regional     4050
Name: TypeMedia, dtype: int64

Quand même 4050 résultats local/régional dans TVA. 

# Création de CSV et Tableau croisée dynamique pour explorer davantage

In [69]:
## Tableau Villes par type_mots-clés 
villes_by_type_motsclés = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="TypeMotClés", aggfunc="count")
villes_by_type_motsclés


TypeMotClés       local  mixte  national
Villes                                  
Acton Vale         2219   1092      1106
Alma               2499   1144      1279
Amos               2537   1182      1239
Amqui              2326   1093       987
Baie-Comeau        2546   1184      1317
...                 ...    ...       ...
Val-des-Sources    2556   1170      1336
Vaudreuil-Dorion   2471   1191      1112
Victoriaville      2445   1180      1087
Ville-Marie        2587   1164      1325
Windsor            2332   1147      1261

[105 rows x 3 columns]

In [50]:
## Tableau Villes par mots-clés
villes_by_motsclés = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="MotsClés", aggfunc="count")
villes_by_motsclés.to_csv("villes_by_motsclés.csv", na_rep="0.0")

In [70]:
## Détail du Type_media selon le type mot-clés
motcles_by_media_type = pd.pivot_table(fulldata, index="TypeMedia", values="domaine", columns="TypeMotClés", aggfunc="count")
motcles_by_media_type

TypeMotClés      local  mixte  national
TypeMedia                              
hors Québec      11180   4345      9260
local/regional  156658  77194     63257
national         61680  31077     47547
non media        17629   5454      2745

In [8]:
## Détail du Type_media selon la Ville
villes_by_media_type = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="TypeMedia", aggfunc="count")
villes_by_media_type.to_csv("villes_by_media_type.csv", na_rep="0.0")

In [12]:
## Détail du Type_media selon le mots-clés
motscles_by_media_type = pd.pivot_table(fulldata, index=["TypeMotClés", "MotsClés"], values="domaine", columns="TypeMedia", aggfunc="count")
motscles_by_media_type.to_csv("motscles_by_media_type.csv", na_rep="0.0")

Ça, c'est intéressant!
On apprend que: 
- la portion média hors-québec ne compte pas pour beaucoup. 
- Pour les mots-clés locaux, mixtes et nationaux, ce sont les médias locaux qui sont les plus fréquents.
    - Cela pourrait s'expliquer par le fait que Radio-Canada compte divers chapeau et est le média le plus populaire. 
- Il y a aussi plus de résultats pour les mots-clés local que national ou mixte, mais c'est normal car plus de mots-clés à la base. 

In [14]:
## CSV qui comprend le nombre de médias le type_media par type de mot-clés par ville ##
type_media_by_ville_by_type_motsclés = pd.pivot_table(fulldata, values="domaine", aggfunc="count", index=["Villes", "TypeMotClés"], columns="TypeMedia")
type_media_by_ville_by_type_motsclés.to_csv("villesparmotscles2.csv", na_rep="0.0")

Document généré, dispo dans le dossier data-analysis. C'est plutôt intéressant. 
On y apprend :
- Montréal_V2 est inopérable. Il manque dans les mots-clés national pour une raison incomprise à date... (l'erreur a été réparée, le fichier de Montreal_V2 national avait une nomenclature différente.) 

In [45]:
villes_by_siege_social = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="VilleSiègeSocial", aggfunc="count")
villes_by_siege_social.to_csv("villes_by_siege_social.csv", na_rep="0.0")


^^ Ce document permet de voir le siège social des médias affichés dans Google News selon la Ville de recherche. 
Par exemple, on voit qu'à Montréal, 4483 des résultats ont un siège social à Montréal. 

In [41]:
### Créer une vue où hors-québec et non media sont retirés. 
data_without_hqc = fulldata[fulldata["TypeMedia"].str.contains("hors Québec") == False]
data_without_hqc_nmedia = data_without_hqc[data_without_hqc["TypeMedia"].str.contains("non media") == False]
data_without_hqc_nmedia

Villes Mots-clés  position du média  \
0                Longueuil  accident                  1   
1            Rouyn-Noranda  accident                  1   
3            Rouyn-Noranda  accident                  2   
5                Longueuil  accident                  3   
7                Longueuil  accident                  4   
...                    ...       ...                ...   
488021  Dolbeau-Mistassini   déficit                  4   
488022  Dolbeau-Mistassini   déficit                  5   
488023  Dolbeau-Mistassini   déficit                  6   
488024  Dolbeau-Mistassini   déficit                  7   
488025  Dolbeau-Mistassini   déficit                  8   

                   Nom du média  \
0        Le Journal de Montréal   
1                  Radio-Canada   
3                     La Presse   
5            Le Courrier du Sud   
7                 TVA Nouvelles   
...                         ...   
488021  Le Quotidien (Saguenay)   
488022  Le Quotidien (Saguenay)   
488023  Le Quotidien (Saguenay)   
488024  Le Quotidien (Saguenay)   
488025  Le Quotidien (Saguenay)   

                                       Titre de l'article Date de publication  \
0       Longueuil: un accident de voiture force la fer...          2021-07-27   
1       Une nouvelle voie de dépassement sur la route 117          2021-07-07   
3       Un mort et 2 blessés dans deux accidents à Rou...          2020-12-09   
5       Un homme de 22 ans perd la vie dans un acciden...          2021-07-17   
7       Spectaculaire accident sur la route 132 en ple...          2021-06-18   
...                                                   ...                 ...   
488021  Un parc multisport de 3,3 M$ sur l'ancien terr...          2021-09-16   
488022  Dolbeau-Mistassini inaugure son café libre-ser...          2021-09-24   
488023  Dolbeau-Mistassini bonifie ses programmes d'ai...          2021-07-13   
488024  Dolbeau-Mistassini veut développer son autonom...          2021-02-15   
488025  La bureaucratie empêche une famille belge de v...          2021-10-03   

        Journée de différence  \
0                         7.0   
1                        27.0   
3                       237.0   
5                        17.0   
7                        46.0   
...                       ...   
488021                   93.0   
488022                   85.0   
488023                  158.0   
488024                  306.0   
488025                   76.0   

                                                      URL  \
0       https://www.journaldemontreal.com/2021/07/26/l...   
1       https://ici.radio-canada.ca/nouvelle/1807406/n...   
3       https://www.lapresse.ca/actualites/justice-et-...   
5       https://www.lecourrierdusud.ca/Un-homme-de-22-...   
7       https://www.tvanouvelles.ca/2021/06/18/spectac...   
...                                                   ...   
488021  https://www.lequotidien.com/2021/09/16/un-parc...   
488022  https://www.lequotidien.com/2021/09/24/dolbeau...   
488023  https://www.lequotidien.com/2021/07/13/dolbeau...   
488024  https://www.lequotidien.com/2021/02/16/dolbeau...   
488025  https://www.lequotidien.com/2021/10/03/la-bure...   

       date et heure saisie type mot-clés                  media_url  \
0       2021-08-03 08:00:09         local  www.journaldemontreal.com   
1       2021-08-03 08:00:09         local        ici.radio-canada.ca   
3       2021-08-03 08:00:09         local            www.lapresse.ca   
5       2021-08-03 08:00:09         local     www.lecourrierdusud.ca   
7       2021-08-03 08:00:09         local        www.tvanouvelles.ca   
...                     ...           ...                        ...   
488021  2021-12-18 17:53:23      national        www.lequotidien.com   
488022  2021-12-18 17:53:23      national        www.lequotidien.com   
488023  2021-12-18 17:53:23      national        www.lequotidien.com   
488024  2021-12-18 17:53:23      national        www.lequotidien.co

In [42]:
data_without_hqc_nmedia.TypeMedia.value_counts()

local/regional    297109
national          140304
Name: Type_media, dtype: int64

# Exploration journée de différence

In [71]:
# La médiane de la Journée de différence selon le type_media
jourdiff_by_type_media_median = pd.pivot_table(fulldata, index="TypeMedia", values="JournéeDifférence", aggfunc=np.median)
jourdiff_by_type_media_median

JournéeDifférence
TypeMedia                        
hors Québec                  22.0
local/regional               92.0
national                     94.0
non media                    19.0

In [72]:
# La moyenne de la Journée de différence selon le type_media
jourdiff_by_type_media_mean = pd.pivot_table(fulldata, index="TypeMedia", values="JournéeDifférence", aggfunc=np.mean)
jourdiff_by_type_media_mean

JournéeDifférence
TypeMedia                        
hors Québec             68.981970
local/regional         134.267529
national               140.657696
non media               51.052942

In [73]:
# La médiane de la Journée de différence selon le type_mots-clés
jourdiff_by_type_motsclés_median = pd.pivot_table(fulldata, index="TypeMotClés", values="JournéeDifférence", aggfunc=np.median)
jourdiff_by_type_motsclés_median

JournéeDifférence
TypeMotClés                   
local                     80.0
mixte                     71.0
national                 103.0

In [74]:
# La moyenne de la Journée de différence selon le type_mots-clés
jourdiff_by_type_motsclés_mean = pd.pivot_table(fulldata, index="TypeMotClés", values="JournéeDifférence", aggfunc=np.mean)
jourdiff_by_type_motsclés_mean

JournéeDifférence
TypeMotClés                   
local               124.373982
mixte               109.461738
national            154.345865

In [76]:
# La moyenne de la Journée de différence selon le domaine
jourdiff_by_domaine_mean = pd.pivot_table(fulldata, index="domaine", values="JournéeDifférence", columns=["TypeMedia","TypeMotClés"], aggfunc=np.mean)
jourdiff_by_domaine_mean.to_csv("jourdiff_by_domaine_mean.csv", na_rep=0.0)

TypeMedia          hors Québec                        local/regional         \
TypeMotClés              local      mixte    national          local  mixte   
domaine                                                                       
2424.mg                   1.00        NaN         NaN            NaN    NaN   
45enord.ca                 NaN        NaN         NaN            NaN    NaN   
45secondes.fr             2.00  12.500000    1.600000            NaN    NaN   
5minutes.rtl.lu          38.75  35.800000  114.333333            NaN    NaN   
7joursinfo.com             NaN        NaN   26.000000            NaN    NaN   
...                        ...        ...         ...            ...    ...   
www.zonebourse.com       39.50  61.172414   51.357143            NaN    NaN   
www.zoomdici.fr          10.00   2.000000    3.000000            NaN    NaN   
xbox-mag.net               NaN        NaN         NaN            NaN    NaN   
zone911.com             262.00        NaN         NaN       17.62963  35.25   
zoom-eco.net               NaN        NaN    3.250000            NaN    NaN   

TypeMedia                      national                        non media  \
TypeMotClés        national       local       mixte   national     local   
domaine                                                                    
2424.mg                 NaN         NaN         NaN        NaN       NaN   
45enord.ca              NaN    7.250000    4.000000  14.222222       NaN   
45secondes.fr           NaN         NaN         NaN        NaN       NaN   
5minutes.rtl.lu         NaN         NaN         NaN        NaN       NaN   
7joursinfo.com          NaN         NaN         NaN        NaN       NaN   
...                     ...         ...         ...        ...       ...   
www.zonebourse.com      NaN         NaN         NaN        NaN       NaN   
www.zoomdici.fr         NaN         NaN         NaN        NaN       NaN   
xbox-mag.net            NaN         NaN         NaN        NaN       NaN   
zone911.com             NaN  117.711252  122.128571  71.444444       NaN   
zoom-eco.net            NaN         NaN         NaN        NaN       NaN   

TypeMedia                          
TypeMotClés        mixte national  
domaine                            
2424.mg              NaN      NaN  
45enord.ca           NaN      NaN  
45secondes.fr        NaN      NaN  
5minutes.rtl.lu      NaN      NaN  
7joursinfo.com       NaN      NaN  
...                  ...      ...  
www.zonebourse.com   NaN      NaN  
www.zoomdici.fr      NaN      NaN  
xbox-mag.net         NaN      0.0  
zone911.com          NaN      NaN  
zoom-eco.net         NaN      NaN  

[1391 rows x 12 columns]

In [77]:
# La moyenne de la Journée de différence selon la ville
jourdiff_by_villes_mean = pd.pivot_table(fulldata, index="Villes", values="JournéeDifférence", columns=["TypeMedia","TypeMotClés"], aggfunc=np.mean)
jourdiff_by_villes_mean.to_csv("jourdiff_by_villes_mean.csv", na_rep=0.0)

TypeMedia        hors Québec                        local/regional  \
TypeMotClés            local      mixte    national          local   
Villes                                                               
Acton Vale        230.750000        NaN   61.666667     152.509131   
Alma               26.165414  20.943182   51.438144     125.906602   
Amos               70.149254  29.783784   98.301775     158.486599   
Amqui             111.333333        NaN  176.333333     169.037364   
Baie-Comeau        17.636364   4.333333  137.400000     138.896947   
...                      ...        ...         ...            ...   
Val-des-Sources    20.473510   9.375000  101.935622     105.738379   
Vaudreuil-Dorion    2.000000   2.000000  375.500000     150.732724   
Victoriaville      22.000000   2.000000   28.909091      87.086548   
Ville-Marie         8.094262   5.982759   14.636364      72.876173   
Windsor           122.905512  95.035316  148.400419     128.075612   

TypeMedia                                   national                          \
TypeMotClés            mixte    national       local       mixte    national   
Villes                                                                         
Acton Vale        154.184810  146.796415  151.989831  202.478788  217.926230   
Alma              119.322738  147.921811  104.040678   98.966443  162.986364   
Amos              106.575092  173.228148  119.943243  131.075099  160.273529   
Amqui             151.589928  283.671827  157.239216  208.155844  386.496774   
Baie-Comeau       110.283224  157.979381  122.206311   99.218750  144.687927   
...                      ...         ...         ...         ...         ...   
Val-des-Sources   105.338750  136.056728  140.226744  127.375610  177.171004   
Vaudreuil-Dorion  146.385671  215.657588  100.403643   79.148230  160.417288   
Victoriaville      85.036923  134.152623   83.090452   88.104348  191.453333   
Ville-Marie        68.571726  103.467312   75.968153   62.704545  103.884726   
Windsor           104.267913  138.006369  105.841549  145.819672  160.602968   

TypeMedia         non media                          
TypeMotClés           local       mixte    national  
Villes                                               
Acton Vale        71.508108  101.680000  131.555556  
Alma              42.726190   47.704545   37.295455  
Amos              42.349057   36.983333   88.333333  
Amqui             61.441103   43.722846  124.406250  
Baie-Comeau       31.819277   17.846154   40.375000  
...                     ...         ...         ...  
Val-des-Sources   67.707865   39.526316   81.450000  
Vaudreuil-Dorion  34.488372   63.846154  108.000000  
Victoriaville     46.481283   36.409836    9.600000  
Ville-Marie       18.304498    8.905263   74.781818  
Windsor           21.589474   43.191176   40.558140  

[105 rows x 12 columns]

# Exploration position du média

In [78]:
# La moyenne de la position du média selon le domaine
positionmedia_by_domaine_mean = pd.pivot_table(fulldata, index="domaine", values="PositionMédia", aggfunc=np.mean)
positionmedia_by_domaine_mean

PositionMédia
domaine                           
2424.mg                  21.000000
45enord.ca               14.807692
45secondes.fr            12.090909
5minutes.rtl.lu          15.176471
7joursinfo.com           12.000000
...                            ...
www.zoomdici.fr          16.857143
www1.alliancefr.com      18.562500
xbox-mag.net              6.000000
zone911.com              12.622611
zoom-eco.net             10.750000

[1444 rows x 1 columns]

In [79]:
# La moyenne de la position du média selon le type_media
positionmedia_by_typemedia_mean = pd.pivot_table(fulldata, index="TypeMedia", values="PositionMédia", columns=["TypeMotClés", "Villes"], aggfunc=np.mean)
positionmedia_by_typemedia_mean

TypeMotClés         local                                               \
Villes         Acton Vale       Alma       Amos      Amqui Baie-Comeau   
TypeMedia                                                                
hors Québec     19.250000  10.375940  15.027778   9.333333   14.818182   
local/regional  12.819104  13.285348  12.880307  12.425868   12.763824   
national        10.925287  12.240854  13.630273  11.517350   13.861432   
non media       12.265957  10.313953  11.269697  13.247031   12.155689   

TypeMotClés                                                                    \
Villes         Baie-Saint-Paul Beauceville Blainville Blanc-Sablon Boischatel   
TypeMedia                                                                       
hors Québec          11.980952    5.000000  12.225455    11.690141   5.000000   
local/regional       13.343829   12.936170  12.875000    11.769231  11.273598   
national             12.159091   12.937500  12.719718    12.849598  11.979894   
non media            11.242424    9.415385  11.066116     7.913043   9.825397   

TypeMotClés     ...       national                                        \
Villes          ... Trois-Pistoles Trois-Rivières Témiscouata-sur-le-Lac   
TypeMedia       ...                                                        
hors Québec     ...      17.714286      14.537313              14.600000   
local/regional  ...      12.049248      12.685435              12.307783   
national        ...      12.318885      13.403226              12.621711   
non media       ...      11.900000      10.700000              13.066667   

TypeMotClés                                                               \
Villes           Val-d'Or Val-des-Monts Val-des-Sources Vaudreuil-Dorion   
TypeMedia                                                                  
hors Québec     14.125000     14.207407       12.485232        19.000000   
local/regional  13.096183     12.513580       13.104534        13.253247   
national        12.841167     11.826840       12.228873        12.131043   
non media       12.210526     13.434783       12.238095        14.187500   

TypeMotClés                                          
Villes         Victoriaville Ville-Marie    Windsor  
TypeMedia                                            
hors Québec        11.909091   11.012987  12.463265  
local/regional     12.533333   13.065060  13.970930  
national           11.539823   13.134286  12.007194  
non media           9.281250   10.285714  11.860465  

[4 rows x 315 columns]

In [80]:
# La moyenne de la position du média selon le type_media
positionmedia_by_typemedia_median = pd.pivot_table(fulldata, index="TypeMedia", values="PositionMédia", columns=["TypeMotClés", "Villes"], aggfunc=np.median)
positionmedia_by_typemedia_median

TypeMotClés         local                                                \
Villes         Acton Vale  Alma  Amos Amqui Baie-Comeau Baie-Saint-Paul   
TypeMedia                                                                 
hors Québec          19.0   9.0  15.0   4.0        18.0            10.0   
local/regional       13.0  14.0  13.0  12.0        13.0            14.0   
national             10.0  12.0  14.0  11.0        15.0            12.0   
non media            12.0   8.5  10.0  13.0        12.0            10.0   

TypeMotClés                                                    ...  \
Villes         Beauceville Blainville Blanc-Sablon Boischatel  ...   
TypeMedia                                                      ...   
hors Québec            4.0       12.0         10.0        5.0  ...   
local/regional        13.0       13.0         11.0       10.0  ...   
national              13.0       13.0         13.0       12.0  ...   
non media              8.0        9.0          5.0        8.0  ...   

TypeMotClés          national                                                 \
Villes         Trois-Pistoles Trois-Rivières Témiscouata-sur-le-Lac Val-d'Or   
TypeMedia                                                                      
hors Québec              20.0           15.0                   13.0     13.0   
local/regional           12.0           13.0                   12.0     13.0   
national                 12.0           14.0                   12.0     13.0   
non media                11.0            9.5                   14.0     10.5   

TypeMotClés                                                                  \
Villes         Val-des-Monts Val-des-Sources Vaudreuil-Dorion Victoriaville   
TypeMedia                                                                     
hors Québec             15.0            12.0             19.0          12.0   
local/regional          12.0            13.0             13.0          12.0   
national                12.0            12.0             12.0          10.0   
non media               12.0            10.0             15.0           8.0   

TypeMotClés                         
Villes         Ville-Marie Windsor  
TypeMedia                           
hors Québec           10.0    13.0  
local/regional        13.0    14.0  
national              13.0    12.0  
non media              9.5    10.0  

[4 rows x 315 columns]

# Base de données avec top 5 en position media seulement

Le but est d'analyser les données, les plus populaire pour avoir une meilleure idée de leur importance. 

In [83]:
fulldata.PositionMédia.value_counts()

1     21351
2     21079
3     20848
4     20663
5     20479
6     20315
7     20153
8     20018
9     19869
10    19748
11    19608
12    19493
13    19377
14    19275
15    19151
16    19056
17    18963
18    18861
19    18772
20    18688
21    18612
22    18523
23    18442
24    18382
25    18300
Name: PositionMédia, dtype: int64

Ici on a le nombre de résultats par position du média ^^

In [90]:
positionmedia_by_typemedia = pd.pivot_table(fulldata, index="PositionMédia", values="domaine", columns=["TypeMotClés","TypeMedia"], aggfunc="count")
positionmedia_by_typemedia

TypeMotClés         local                                         mixte  \
TypeMedia     hors Québec local/regional national non media hors Québec   
PositionMédia                                                             
1                     323           6635     2905       811         121   
2                     475           6505     2724       870         148   
3                     424           6428     2736       887         149   
4                     445           6440     2627       879         163   
5                     478           6352     2639       838         187   
6                     476           6340     2603       823         198   
7                     453           6356     2539       821         212   
8                     477           6302     2519       817         217   
9                     484           6369     2408       795         159   
10                    460           6285     2514       747         179   
11                    481           6272     2481       705         183   
12                    445           6285     2446       699         176   
13                    491           6233     2409       681         173   
14                    443           6250     2399       687         172   
15                    453           6193     2430       648         167   
16                    438           6217     2382       638         187   
17                    414           6140     2476       601         176   
18                    450           6211     2312       618         169   
19                    448           6213     2273       621         203   
20                    454           6200     2276       593         174   
21                    419           6118     2369       578         166   
22                    410           6098     2332       602         187   
23                    433           6072     2327       569         173   
24                    461           6047     2304       563         156   
25                    445           6097     2250       538         150   

TypeMotClés                                        national                 \
TypeMedia     local/regional national non media hors Québec local/regional   
PositionMédia                                                                
1                       3308     1319       254         355           2960   
2                       3175     1371       276         347           3006   
3                       3152     1325       303         355           2831   
4                       3124     1338       279         378           2850   
5                       3178     1230       280         380           2775   
6                       3126     1260       267         364           2722   
7                       3084     1284       259         348           2685   
8                       3073     1271       252         370           2663   
9                       3112     1281       239         344           2617   
10                      3097     1248       249         366           2591   
11                      3102     1222       242         351           2540   
12                      3067     1272       213         371           2460   
13                      3077     1258       203         354           2525   
14                      3097     1215       207         347           2463   
15                      3013     1287       203         356           2462   
16                      3054     1220       192         370           2423   
17                      3054     1214       194         374           2326   
18                      3057     1219       168         373           2376   
19                      3025     1175       192         395           2326   
20                      3047     1180       177         385           2337   
21                      3012     1193       192         396           2280   
22                      304

Intéressant, on remarque que dans l'ensemble, les médias locaux/régionaux sont plus nombreux peu importe le mot-clé et sa catégorie. 

# Coefficient de Gini

In [19]:
## Pivot table qui comprend le nombre de médias le type_media par type de mot-clés par ville ##
type_media_by_ville_by_type_motsclés2 = pd.pivot_table(fulldata, values="domaine", aggfunc="count", columns=["Villes", "TypeMotClés"], index="TypeMedia", fill_value=0.0)
type_media_by_ville_by_type_motsclés2

Villes         Acton Vale                 Alma                 Amos        \
TypeMotClés         local mixte national local mixte national local mixte   
TypeMedia                                                                   
hors Québec             8     0        6   133    88      209   108    37   
local/regional       1675   878      935  1952   851      796  1696   819   
national              348   187      155   328   161      230   403   266   
non media             188    27       10    86    44       44   330    60   

Villes                  Amqui  ... Vaudreuil-Dorion Victoriaville        \
TypeMotClés    national local  ...         national         local mixte   
TypeMedia                      ...                                        
hors Québec         170     3  ...                2            12     2   
local/regional      682  1585  ...              308          1986   996   
national            356   317  ...              786           248   121   
non media            31   421  ...               16           199    61   

Villes                  Ville-Marie                Windsor                 
TypeMotClés    national       local mixte national   local mixte national  
TypeMedia                                                                  
hors Québec          11         244    58      154     382   538      490  
local/regional      705        1072   481      415    1467   346      172  
national            339         982   530      700     293   195      556  
non media            32         289    95       56     190    68       43  

[4 rows x 315 columns]

In [8]:
def gini(x):
    total = 0
    for i, xi in enumerate(x[:-1], 1):
        total += np.sum(np.abs(xi - x[i:]))
    return total / (len(x)**2 * np.mean(x))

164.71911844860725